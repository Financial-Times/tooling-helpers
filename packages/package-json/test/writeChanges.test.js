const loadPackageJson = require("../src/index.js");

const { fs, vol } = require('memfs');
const actualFs = jest.requireActual('fs');

jest.mock('fs', () => require('memfs').fs);

let mockFsWriteFileSync;

beforeEach(() => {
    mockFsWriteFileSync = jest.spyOn(fs, 'writeFileSync');

    const fixturePackageJson = `${__dirname}/fixtures/testPackageJson.json`;
    const volume = {
        'package.json': actualFs.readFileSync(fixturePackageJson, { encoding: 'utf-8' })
    };
    vol.fromJSON(volume);
});

afterEach(() => {
    mockFsWriteFileSync.mockRestore();
    jest.clearAllMocks();
});

describe.only("writeChanges", () => {

  test("writes updated package.json document to the file system", () => {

    const packageJsonBefore = loadPackageJson({
      filepath: 'package.json'
    });
    
    packageJsonBefore.removeDependency({
      pkg: "prettier",
      field: "devDependencies"
    });

    packageJsonBefore.writeChanges();
    expect(mockFsWriteFileSync).toHaveBeenCalled();

    const packageJsonAfter = loadPackageJson({
        filepath: 'package.json'
    });

    const devDependenciesAfter = packageJsonAfter.getDocument().devDependencies;
    expect(devDependenciesAfter.prettier).toEqual(undefined);

    const packageJsonDocumentBefore = packageJsonBefore.getDocument();
    const packageJsonDocumentAfter = packageJsonAfter.getDocument();
    expect(packageJsonDocumentBefore).toEqual(packageJsonDocumentAfter);
  });

});
