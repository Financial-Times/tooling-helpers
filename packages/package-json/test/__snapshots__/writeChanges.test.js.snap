// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeChanges throws an error if it is unable to write to the package.json file 1`] = `
"package-json#writeChanges: Error writing changes to file '/app/package.json'

EACCES: permission denied, open '/app/package.json'"
`;
